app/models/mixins/has_network_manager_mixin.rb

```diff
@@ -22,14 +20,15 @@ module HasManyCloudNetworksMixin
              :to        => :network_manager,
              :allow_nil => true
 
-    before_validation :ensure_managers
+    alias_method :all_cloud_networks, :cloud_networks
```

Comment:
I'm a bit confused about `all_cloud_networks`
It looks like this is always the same as `cloud_networks` for `NetworkManager` and its only really used in `CloudTenant`
Can't you remove `all_cloud_networks` then?